osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs

```diff
@@ -53,10 +52,29 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            scoreProcessor.OnResetFromReplayFrame += updateAllCounts;
             scoreProcessor.NewJudgement += judgement => updateCount(judgement, false);
             scoreProcessor.JudgementReverted += judgement => updateCount(judgement, true);
         }
 
+        private bool hasUpdatedCounts;
+
+        private void updateAllCounts()
```

Comment:
I don't feel the best about these being named generically even though they're really specific to the "reset from replay frame" case. How would you feel about the following renames?

```diff
diff --git a/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs b/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs
index 2562e26127..c00cb3487b 100644
--- a/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs
+++ b/osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs
@@ -52,16 +52,16 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            scoreProcessor.OnResetFromReplayFrame += updateAllCounts;
+            scoreProcessor.OnResetFromReplayFrame += updateAllCountsFromReplayFrame;
             scoreProcessor.NewJudgement += judgement => updateCount(judgement, false);
             scoreProcessor.JudgementReverted += judgement => updateCount(judgement, true);
         }
 
-        private bool hasUpdatedCounts;
+        private bool hasUpdatedCountsFromReplayFrame;
 
-        private void updateAllCounts()
+        private void updateAllCountsFromReplayFrame()
         {
-            if (hasUpdatedCounts)
+            if (hasUpdatedCountsFromReplayFrame)
                 return;
 
             foreach (var kvp in scoreProcessor.Statistics)
@@ -72,7 +72,7 @@ private void updateAllCounts()
                 count.ResultCount.Value = kvp.Value;
             }
 
-            hasUpdatedCounts = true;
+            hasUpdatedCountsFromReplayFrame = true;
         }
 
         private void updateCount(JudgementResult judgement, bool revert)

```

Reply:
Seems fine, applied.